elasticapm/context/base.py

```diff
@@ -31,13 +31,48 @@
 
 class BaseContext(object):
     def set_transaction(self, transaction):
+        """
+        Set the transaction for the current execution context
+        """
         raise NotImplementedError
 
     def get_transaction(self, clear=False):
+        """
+        Get the transaction for the current execution context
+
+        If clear=True, also set the transaction to None for the current
+        execution context.
+        """
+        raise NotImplementedError
+
+    def set_span(self, span, extra=None):
+        """
+        Set the active span for the current execution context.
+
+        The previously-activated span will be saved to be re-activated later.
+
+        Optionally, `extra` data can be provided and will be saved alongside
+        the span.
+        """
         raise NotImplementedError
 
-    def set_span(self, span):
+    def get_span(self, extra=False):
```

Comment:
I think it would help with legibility if the `extra` argument had a different name here compared to `set_span`, as they have a different meaning. Something like `with_extra`. Same in `unset_span`.